### PR TITLE
 migrate from databaseUsername to databaseAccount and fully use MariaDBAccount

### PR DESCRIPTION
**Pull request by zzzeek**
Lead Jira: [OSPRH-4095](https://issues.redhat.com/browse/OSPRH-4095)

1. [x] controller calls EnsureMariaDBAccount up front to make sure MariaDBAccount is present
2. [x] error code from EnsureMariaDBAccount is handled, Conditions are amended when error is returned
3. [x] controller calls NewDatabaseForAccount instead of NewDatabase
4. [x]  GetAccountAndSecret is used to retrieve account /secret to populate template
5. [x]  GetDatabaseByName() , normally used for delete finalizers, replaced with GetDatabaseByNameAndAccount
6. [x]  CreateOrPatchAll() used to patch the Database, replacing CreateOrPatchDB / CreateOrPatchDBByName
7. [x]  controller calls DeleteUnusedMariaDBAccountFinalizers when launched pods are definitely running on a new MariaDBAccount, returns error code if present
8. [x]  PasswordSelectors that refer to database are removed
9. [x]  all databaseUser replaced with databaseAccount inside of all XYZ_types.go
10. [x] all databaseUser replaced with databaseAccount inside of all `kuttl/*.yaml`
11. [x] all default databaseUser names become databaseAccount, replacing underscores with dashes inside of all XYZ_types.go
12. [x] all default databaseUser names become databaseAccount, replacing underscores with dashes inside of all `kuttl/*.yaml`
13. [x] MariaDBAccountSuiteTests are used in controller ginkgo tests if it has them
14. [x] [184](https://github.com/openstack-k8s-operators/mariadb-operator/pull/184) is merged and replaces from go.mod are removed

### CODE REVIEW

File: config/manager/kustomization.yaml [hunk 1]
```diff
@@ -12,5 +12,5 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
 - name: controller
-  newName: quay.io/openstack-k8s-operators/neutron-operator
```
Comment (zzzeek) [inline config/manager/kustomization.yaml:15]: need to revert this

File: config/manifests/bases/neutron-operator.clusterserviceversion.yaml [hunk 2]
```diff
@@ -18,6 +18,10 @@ spec:
       displayName: Neutron API
       kind: NeutronAPI
       name: neutronapis.neutron.openstack.org
+      specDescriptors:
```
Comment (zzzeek) [inline config/manifests/bases/neutron-operator.clusterserviceversion.yaml:21]: im not sure why this got here, I would assume when this was last updated to the most recent mariadb this wasn't built (is this "make bundle"?)

**Review by karelyatin — COMMENTED** [review]
lgtm

few cleanups inline

File: test/kuttl/common/assert_sample_deployment.yaml [hunk 3]
```diff
@@ -17,9 +17,8 @@ spec:
     [DEFAULT]
     debug = true
   databaseInstance: openstack
-  databaseUser: neutron
+  databaseAccount: neutron
   passwordSelectors:
-    database: NeutronDatabasePassword
```
[the remark this answers is not in the document]
Comment (zzzeek) [reply test/kuttl/common/assert_sample_deployment.yaml:22]: yah let me do that, neutron seems to not have too many problems getting through CI

**Comment by zzzeek** [comment]
recheck

**Comment by zzzeek** [comment]
Ok it's ready ready ready

**Review by karelyatin — APPROVED** [review]
/lgtm

**Pull request merged**